src/libValidator/Validator.cpp

```diff
@@ -293,6 +294,11 @@ bool Validator::CheckDirBlocks(
       BlockStorage::GetBlockStorage().PutDSBlock(
           dsblock.GetHeader().GetBlockNum(), serializedDSBlock);
       m_mediator.m_node->UpdateDSCommiteeComposition(mutable_ds_comm, dsblock);
+      if (ENABLE_INCR_DB) {
```

Comment:
Why insert into the DB again here during the checking? Isn't redundant? Because I only saw this func called after finishing the fetching data from Incremental db

Reply:
`CheckDirBlocks` is called at many places except here.

Reply:
I know, but seems when you call this during `IncrementalDB::VerifyAll` it doesn't need to redo this insertion to the Incremental db again for the DS Block